apicast/src/configuration_loader/remote_v2.lua

```diff
@@ -203,12 +218,59 @@ function _M:services()
   if res.status == 200 then
     local json = cjson.decode(res.body)
 
-    return json.services or {}
+    return json.services or array()
   else
     return nil, status_code_error(res)
   end
 end
 
+local function openid_configuration_url(endpoint)
+  return resty_url.join(endpoint, '.well-known/openid-configuration')
+end
+
+function _M:oidc_issuer_configuration(service)
+  local http_client = self.http_client
+
+  if not http_client then
+    return nil, 'not initialized'
+  end
+
+  local endpoint = service.oidc.issuer_endpoint
+
+  if not endpoint then
+    return nil, 'no OIDC endpoint'
+  end
+
+  local uri = openid_configuration_url(endpoint)
+  local res = http_client.get(uri)
+
+  if not res.ok then
+    -- TODO: log the response
+    return nil, 'could not get OpenID Connect configuration'
+  end
+
+  local config = res.headers.content_type == 'application/json' and cjson.decode(res.body)
+
+  if not config then
+    ngx.log(ngx.STDERR, 'failed to get OIDC Issuer Configuration from ', uri, ' status: ', res.status, ' body: ', res.body)
```

Comment:
I'd suggest to change the log to `ERR` and move this to the TODO a few lines above.
And here log explicitly that that the config couldn't be parsed because the content is not a valid JSON.